api/utils/keys/hardwarekey/hardwarekey.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2025 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package hardwarekey defines types and interfaces for hardware private keys.
+package hardwarekey
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"encoding/json"
+	"io"
+
+	"github.com/gravitational/trace"
+)
+
+// Service for interfacing with hardware private keys.
+type Service interface {
+	// NewPrivateKey creates or retrieves a hardware private key for the given config.
+	NewPrivateKey(ctx context.Context, config PrivateKeyConfig) (*PrivateKey, error)
+	// Sign performs a cryptographic signature using the specified hardware
+	// private key and provided signature parameters.
+	Sign(ctx context.Context, ref *PrivateKeyRef, rand io.Reader, digest []byte, opts crypto.SignerOpts) (signature []byte, err error)
+	// GetFullKeyRef gets the full [PrivateKeyRef] for an existing hardware private
+	// key in the given slot of the hardware key with the given serial number.
+	GetFullKeyRef(serialNumber uint32, slotKey PIVSlotKey) (*PrivateKeyRef, error)
+}
+
+// PrivateKey is a hardware private key implementation of [crypto.Signer].
+type PrivateKey struct {
```

Comment:
I don't love the name `PrivateKey` because this is not actually the private key - the private key never leaves the hardware device. Should this be named `hardwarekey.Signer` instead?